### PR TITLE
Allow 1 as target goal 150

### PR DESCRIPTION
**Pull request by eklecky**
Add code to allow 1 as the target book/pages goal.
Add test class called ReadingGoalFormTests to test ReadingGoalForm and confirm that fix was successful.

https://github.com/knjk04/book-project/issues/150

Bug Fix screenshot 0

Bug Fix screenshot 1

### CODE REVIEW

File: src/test/java/com/karankumar/bookproject/ui/goal/ReadingGoalFormTests.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
+package com.karankumar.bookproject.ui.goal;
+
+
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/ui/goal/ReadingGoalFormTests.java:3]: Could you delete this extra line?

File: src/test/java/com/karankumar/bookproject/ui/goal/ReadingGoalFormTests.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.ApplicationContext;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.context.web.WebAppConfiguration;
+
+import java.util.Random;
+
+
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/ui/goal/ReadingGoalFormTests.java:24]: Could you delete this extra line?

File: src/test/java/com/karankumar/bookproject/ui/goal/ReadingGoalFormTests.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+@ExtendWith(SpringExtension.class)
+@SpringBootTest
+@WebAppConfiguration
+
+public class ReadingGoalFormTests {
+     static Routes routes;
+
+
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/ui/goal/ReadingGoalFormTests.java:32]: another extra line

File: src/test/java/com/karankumar/bookproject/ui/goal/ReadingGoalFormTests.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    @BeforeEach
+    public void setup(@Autowired ReadingGoalService goalService, @Autowired PredefinedShelfService predefinedShelfService) {
+        final SpringServlet servlet = new MockSpringServlet(routes, ctx);
+        MockVaadin.setup(UI::new, servlet);
+
+        Assumptions.assumeTrue(goalService != null);
+        goalService.deleteAll(); // reset
+
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/ui/goal/ReadingGoalFormTests.java:48]: another extra line

File: src/test/java/com/karankumar/bookproject/ui/goal/ReadingGoalFormTests.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/ui/goal/ReadingGoalFormTests.java:51]: another extra line

File: src/test/java/com/karankumar/bookproject/ui/goal/ReadingGoalFormTests.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    private ReadingGoal.GoalType getRandomGoalType() {
+        ReadingGoal.GoalType[] goalTypes = ReadingGoal.GoalType.values();
+        return goalTypes[new Random().nextInt(goalTypes.length)];
+    }
+
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/ui/goal/ReadingGoalFormTests.java:56]: another extra line

File: src/test/java/com/karankumar/bookproject/ui/goal/ReadingGoalFormTests.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+            ReadingGoalForm goalForm = new ReadingGoalForm();
+
+            goalForm.targetToRead.setValue(1);
+            goalForm.chooseGoalType.setValue(getRandomGoalType());
+            goalForm.saveButton.click();
+            Assertions.assertTrue(goalForm.binder.isValid());
+
+            goalForm.targetToRead.setValue(0);
+            Assertions.assertFalse(goalForm.binder.isValid());
+
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/ui/goal/ReadingGoalFormTests.java:73]: There should be no extra lines before the closing brace in this method

File: src/test/java/com/karankumar/bookproject/ui/goal/ReadingGoalFormTests.java [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+        }
+
+        @AfterEach
+        public void tearDown () {
+            MockVaadin.tearDown();
+        }
+
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/ui/goal/ReadingGoalFormTests.java:81]: delete extra line

File: src/test/java/com/karankumar/bookproject/ui/goal/ReadingGoalViewTests.java [hunk 9]
```diff
@@ -78,7 +78,6 @@ public void progressValueCorrect() {
         // ensure 0, and not an arithmetic exception, is returned
         Assertions.assertEquals(ReadingGoalView.getProgress(0, 5), 0);
     }
-
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/ui/goal/ReadingGoalViewTests.java:81]: This line was needed (we have one newline separating each method)

**Comment by knjk04** [comment]
Thanks for this! The test looks the same as the one I had in mind

**Comment by eklecky** [comment]
No problem. I'll remove the extra lines now.

File: src/main/java/com/karankumar/bookproject/ui/goal/ReadingGoalForm.java [hunk 10]
```diff
@@ -77,7 +78,7 @@ private IntegerField createBooksGoalField() {
     /**
      * @return a save button
      */
-    private Button createSaveButton() {
```
[the remark this answers is not in the document]
Comment (eklecky) [reply src/main/java/com/karankumar/bookproject/ui/goal/ReadingGoalForm.java]: Done

**Pull request merged**